### PR TITLE
Clarify docs of the `filter` argument in `MessageHandler`

### LINKED ISSUE #3394: [FEATURE] Switch `filters` to kwarg in `MessageHandler`
### What kind of feature are you missing? Where do you notice a shortcoming of PTB?

The `filters` argument of the `MessageHandler` constructor is currently a positional argument and defaults to `filters.ALL` if `None` is explicitly passed:

https://github.com/python-telegram-bot/python-telegram-bot/blob/637cc574ab2eab39055a2bd08827c67e9503cd7a/telegram/ext/_messagehandler.py#L82-L90

### Describe the solution you'd like

Using a keyword argument instead:

```python
 def __init__( 
     self,
     callback: HandlerCallback[Update, CCT, RT],
     filters: filters_module.BaseFilter = filters.ALL,
     block: DVInput[bool] = DEFAULT_TRUE,
 ): 
  
     super().__init__(callback, block=block) 
     self.filters = filters
```

Or something similar.

Documentation could be improved as well:

https://github.com/python-telegram-bot/python-telegram-bot/blob/637cc574ab2eab39055a2bd08827c67e9503cd7a/telegram/ext/_messagehandler.py#L47-L51

Seems unclear to me when it is stating `pass ``~filters.UpdateType.*`` in the filter argument. `

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Viicos**
Fixes #3394.

### CODE REVIEW

File: telegram/ext/_messagehandler.py [hunk 1]
```diff
@@ -44,11 +44,8 @@ class MessageHandler(BaseHandler[Update, CCT]):
         filters (:class:`telegram.ext.filters.BaseFilter`): A filter inheriting from
             :class:`telegram.ext.filters.BaseFilter`. Standard filters can be found in
             :mod:`telegram.ext.filters`. Filters can be combined using bitwise
-            operators (& for and, | for or, ~ for not). This defaults to all message updates
-            being: :attr:`telegram.Update.message`, :attr:`telegram.Update.edited_message`,
-            :attr:`telegram.Update.channel_post` and :attr:`telegram.Update.edited_channel_post`.
-            If you don't want or need any of those pass ``~filters.UpdateType.*`` in the filter
-            argument.
+            operators (& for and, | for or, ~ for not). Passing ``None`` is a shortcut
```
Comment (Bibo-Joshi) [inline telegram/ext/_messagehandler.py:47]:
```suggestion
            operators (& for and, | for or, ~ for not). Passing :obj:`None` is a shortcut
```

**Review by Bibo-Joshi — COMMENTED** [review]
Thanks for the PR! I just have a small comment :)

File: telegram/ext/_messagehandler.py (hunk 1 above)
Comment (Viicos) [reply telegram/ext/_messagehandler.py]: Oops, fixed.

**Comment by Bibo-Joshi** [comment]
Thank you for the contribution!

**Pull request merged**